pytorch_lightning/loops/batch/manual.py

```diff
@@ -0,0 +1,93 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, Tuple
+
+from pytorch_lightning.loops import Loop
+from pytorch_lightning.loops.utilities import (
+    _build_training_step_kwargs,
+    _check_training_step_output,
+    _process_training_step_output,
+)
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+
+
+class ManualOptimization(Loop):
+    """
+    A special loop implementing what is known in Lightning as Manual Optimization where the optimization happens
+    entirely in the :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step` and therefore the user
+    is responsible for back-propagating gradients and making calls to the optimizers.
+
+    This loop is a trivial case because it performs only a single iteration (calling directly into the module's
+    :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step`) and passing through the output(s).
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._done: bool = False
+        self._hiddens: Optional[Any] = None
+        self._output: Optional[ResultCollection] = None
+
+    @property
+    def done(self) -> bool:
+        return self._done
+
+    def reset(self) -> None:
+        self._done = False
+
+    def advance(self, batch: Any, batch_idx: int, hiddens: Optional[Any] = None) -> None:  # type: ignore [override]
```

Comment:
```suggestion
    def advance(self, batch: Any, batch_idx: int, hiddens: Optional[Any] = None) -> None:  # type: ignore[override]
```
@carmocca IDE shows me an error with the space

Reply:
You are right, i've been writing it wrong the whole time then 😂